### ISSUE TITLE
MA0124: Does not handle nullable strings

### ISSUE THREAD

**Issue by FreeYourMinds** [meziantou/Meziantou.Analyzer#630]
### Version of the Meziantou.Analyzer NuGet package

2.0.100

### Rule Identifier

MA0124

### Target Framework

net7.0

### C# Language version

C# 11

### Description

# MA0124: Does not handle nullable strings

When configuration contains a type for a nullable string, it still gets reported as wrong type

>  Error MA0124 : Parameter 'NullableString' must be of type 'string?' but is of type 'string?'

## LoggerParameterTypes.txt

```
NullableString;System.Nullable{System.String}
```

### Reproduction Steps

Minimal code:

```c#
var nullableString = (string?)"nullable string";
logger.LogDebug("This {NullableString} raises MA0124", nullableString);
```

### Other information

Using 

```
NullableString;System.String
```

makes MA0124 accept the Parameter as valid

**Comment by meziantou** [comment]
`System.Nullable{System.String}` is not a valid type in .NET. `System.Nullable<T>` only supports structs and `string` is a class.

In your case, you can use `NullableString;System.String` to accept `string` and `string?`.

**Comment by FreeYourMinds** [comment]
I Understand, but still think it would be helpful to point that out in the example for MA0124 and also change the Message for this certain case. The current `must be of type 'string?' but is of type 'string?'` is quite irritating

And shame on me, just copied and pasted from another nullable type in the configuration, not thinking much about it. But I expect this to be the _natural_ developer lazyness 🙈 

Also might be an option to have _MA0125_ report `Systen.Nullable{Sytem.String}` as invalid. 

Could help preventing this little pitfall for others in the future

**Comment by meziantou** [comment]
If you use a recent version of Roslyn, the error message should be better. The type will be fully qualified and expanded.
